cms/djangoapps/contentstore/tests/test_course_settings.py

```diff
@@ -1106,3 +1109,99 @@ def test_add(self):
+    EDITABLE_TIME_FIELD = "<input type=\"text\" class=\"time end\" \
+id=\"course-enrollment-end-time\" value=\"\" placeholder=\"HH:MM\" autocomplete=\"off\"  />"
+
+    def setUp(self):
+        """ Initialize course used to test enrollment fields. """
+        super(CourseEnrollmentEndFieldTest, self).setUp()
+        self.course = CourseFactory.create(org='edX', number='dummy', display_name='Marketing Site Course')
+        self.course_details_url = reverse_course_url('settings_handler', unicode(self.course.id))
+
+    def _get_course_details_response(self, global_staff):
+        """ Return the course details page as either global or non-global staff"""
+        user = UserFactory(is_staff=global_staff)
+        CourseInstructorRole(self.course.id).add_users(user)
+
+        self.client.login(username=user.username, password='test')
+
+        return self.client.get_html(self.course_details_url)
+
+    def _verify_editable(self, response):
+        """
+        Assert that all editable field content exists and no
+        uneditable field content exists for enrollment end fields.
+        """
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, self.NOT_EDITABLE_HELPER_MESSAGE)
+        self.assertNotContains(response, self.NOT_EDITABLE_DATE_WRAPPER)
+        self.assertNotContains(response, self.NOT_EDITABLE_TIME_WRAPPER)
+        self.assertNotContains(response, self.NOT_EDITABLE_DATE_FIELD)
+        self.assertNotContains(response, self.NOT_EDITABLE_TIME_FIELD)
+        self.assertContains(response, self.EDITABLE_DATE_WRAPPER)
+        self.assertContains(response, self.EDITABLE_TIME_WRAPPER)
+        self.assertContains(response, self.EDITABLE_DATE_FIELD)
+        self.assertContains(response, self.EDITABLE_TIME_FIELD)
+
+    def _verify_not_editable(self, response):
+        """
+        Assert that all uneditable field content exists and no
+        editable field content exists for enrollment end fields.
+        """
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.NOT_EDITABLE_HELPER_MESSAGE)
+        self.assertContains(response, self.NOT_EDITABLE_DATE_WRAPPER)
+        self.assertContains(response, self.NOT_EDITABLE_TIME_WRAPPER)
+        self.assertContains(response, self.NOT_EDITABLE_DATE_FIELD)
+        self.assertContains(response, self.NOT_EDITABLE_TIME_FIELD)
+        self.assertNotContains(response, self.EDITABLE_DATE_WRAPPER)
+        self.assertNotContains(response, self.EDITABLE_TIME_WRAPPER)
+        self.assertNotContains(response, self.EDITABLE_DATE_FIELD)
+        self.assertNotContains(response, self.EDITABLE_TIME_FIELD)
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {'ENABLE_MKTG_SITE': False})
+    def test_course_details_with_disabled_setting_global_staff(self):
+        """Test that user enrollment end date is editable in response
```

Comment:
Nit: the first line of a conforming docstring should be a short summary of the method and it shouldn't wrap. 

See: https://github.com/edx/edx-platform/wiki/Python-Guidelines

In particular, we use the so-called Google Style: https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Comments#Comments